my-app/src/components/Board.js

```diff
@@ -1,23 +1,17 @@
-import React from 'react';
+import * as React from 'react';
 import Square from './Square';
+import {GameContext, useHistory, useTakeTurn} from '../game-context';
 
-const Board =({squares, onClick}) => (
-<div>
-    <div className="board-row">
-        <Square value={squares[0]} onClick={()=>{onClick(0)}}/>
-        <Square value={squares[1]} onClick={()=>{onClick(1)}}/>
-        <Square value={squares[2]} onClick={()=>{onClick(2)}}/>
-    </div>
-    <div className="board-row">
-        <Square value={squares[3]} onClick={()=>{onClick(3)}}/>
-        <Square value={squares[4]} onClick={()=>{onClick(4)}}/>
-        <Square value={squares[5]} onClick={()=>{onClick(5)}}/>
-    </div>
-    <div className="board-row">
-        <Square value={squares[6]} onClick={()=>{onClick(6)}}/>
-        <Square value={squares[7]} onClick={()=>{onClick(7)}}/>
-        <Square value={squares[8]} onClick={()=>{onClick(8)}}/>
-    </div>
-</div>
-)
+const Board = () => {
+    const takeTurn = useTakeTurn();
```

Comment:
can you move `useTakeTurn` into `Square` component? You'll also need to update the hook to be able to work like below:

```js
const takeTurn = useTakeTurn(index);
// ...
<button className="square" onClick={takeTurn}>
// ...
```

Reply:
`useTakeTurn` function doesn't get any arguments, you'll need to implement it to satisfy. And it should be work like the previous comment not like `<button className="square" onClick={() => takeTurn(index)}>`